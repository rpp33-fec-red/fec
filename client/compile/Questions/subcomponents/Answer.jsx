import React from 'react';

class Answer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      helpful: false,
      helpfulnessVoteCount: this.props.answer.helpfulness,
      reported: false
    }
    this.handleHelpfulnessVote = this.handleHelpfulnessVote.bind(this);
    this.handleReport = this.handleReport.bind(this);
  }

  handleHelpfulnessVote(event) {
    if  (!this.state.helpful) {
      this.setState({
        helpful: true,
        helpfulnessVoteCount: this.state.helpfulnessVoteCount + 1
      });
    }
  }

  handleReport(event) {
    if (!this.state.reported) {
      this.setState({
        reported: true
      })
    }
  }

  render() {
    return (
      <div className="answer">
        <p className="answerTitle">{this.props.answer.body}</p>
        <div className="answerLinks">
          <p>by {this.props.answer.answerer_name}, MONTH, DD, YEAR | Helpful?
            <a onClick={this.handleHelpfulnessVote}>Yes</a>
            ({this.state.helpfulnessVoteCount}) | &nbsp;
            {!this.state.reported ?
              <a onClick={this.handleReport}>Report</a> :
              <a onClick={this.handleReport}>Reported</a>
            }
          </p>
        </div>
      </div>
    );
  }
}

export default Answer;