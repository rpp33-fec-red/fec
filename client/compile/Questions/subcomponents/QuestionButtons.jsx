import React from 'react';

class QuestionButtons extends React.Component {
  constructor(props) {
    super(props);
    this.state = {

    }
  }

  render() {
    console.log(this.props.allQuestionsDisplayed);
    return (
      <div className="question-buttons">
        {!this.props.allQuestionsDisplayed &&
          <button>MORE ANSWERED QUESTIONS</button>
        }
        <button>ADD A QUESTION +</button>
      </div>
    );
  }
}

export default QuestionButtons;