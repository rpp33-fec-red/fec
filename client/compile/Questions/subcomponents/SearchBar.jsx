import React from 'react';
import PropTypes from 'prop-types';

function SearchBar(props) {
  return (
    <div className="search-bar">
      <input
        type="search"
        id="questions-search-bar"
        placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..."
        onChange={props.search}
      />
    </div>
  );
}

SearchBar.propTypes = {
  search: PropTypes.func
};

export default SearchBar;