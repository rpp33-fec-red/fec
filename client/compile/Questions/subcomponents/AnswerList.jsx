import React from 'react';
import Answer from './Answer.jsx';

class AnswerList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      displayAllAnswers: false,
      answerData: this.props.answers
    }

    this.createAnswerDataArray = this.createAnswerDataArray.bind(this);
  }

  createAnswerDataArray(objectData) {
    let answers = [];
    for (const answer in this.state.answerData) {
      answers.push(this.state.answerData[answer]);
    }
    return answers;
  }

  componentDidMount() {
    this.setState({
      answerData: this.createAnswerDataArray(this.state.answerData)
    })
  }

  render() {
    if (this.state.answerData.length) {
      if (!this.state.displayAllAnswers) {
        this.state.answerData = this.state.answerData.slice(0, 2);
      }

      return(
        <div className="answers">
        <p>A:&nbsp;</p>
        <div className="answerList" >
          {this.state.answerData.map((answer) =>
            <Answer key={answer.id} answer={answer} />
          )}
          {this.state.answerData.length > 2 &&
            <a>See more answers</a>
          }
        </div>
      </div>
      );
    } else {
      return null;
    }
  }
}

export default AnswerList;