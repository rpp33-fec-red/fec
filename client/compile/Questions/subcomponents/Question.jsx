import React from 'react';
import axios from 'axios';
import PropTypes from 'prop-types';
import AnswerList from './AnswerList.jsx';
import AddAnswerModal from '../modals/AddAnswer.jsx';

class Question extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      helpful: false,
      helpfulnessVoteCount: this.props.question.question_helpfulness,
      showAddAnswerModal: false,
      fields: ''
    };
    this.handleHelpfulnessVote = this.handleHelpfulnessVote.bind(this);
    this.handleAddAnswer = this.handleAddAnswer.bind(this);
    this.handleSubmitAnswer = this.handleSubmitAnswer.bind(this);
    this.closeModal = this.closeModal.bind(this);
  }

  handleHelpfulnessVote(event) {
    event.preventDefault();
    let request = {
      data: null,
      endpoint: `/qa/questions/${this.props.question.question_id}/helpful`,
      params: {
        question_id: this.props.question.question_id
      }
    };

    if  (!this.state.helpful) {
      this.setState({
        helpful: true,
        helpfulnessVoteCount: this.state.helpfulnessVoteCount + 1
      }, function() {
        axios.put('/putData', request);
      });
    }
  }

  handleAddAnswer(event) {
    event.preventDefault();
    this.setState({
      showAddAnswerModal: true,
      missingFields: ''
    });
  }

  handleSubmitAnswer(event) {
    event.preventDefault();
    const request = {
      data: {
        body: event.target.answer.value,
        name: event.target.nickname.value,
        email: event.target.email.value,
        photos: [],
      },
      endpoint: `/qa/questions/${this.props.question.question_id}/answers`,
      params: {
        question_id: this.props.question.question_id
      }
    };
    const fields = {
      answer: request.data.body,
      nickname: request.data.name,
      email: request.data.email
    };

    if (fields.answer && fields.nickname && fields.email) {
      this.setState({
        showAddAnswerModal: false
      }, function() {
        axios.post('/postData', request);
      });
    } else {
      let missingFields = '';
      for (var field in fields) {
        if (!fields[field]) {
          missingFields = missingFields + field + ' ';
        }
        this.setState({
          missingFields: missingFields
        });
      }
    }
  }

  closeModal() {
    this.setState({
      showAddAnswerModal: false
    });
  }

  render() {
    return (
      <div className="question">
        {this.state.showAddAnswerModal &&
          <AddAnswerModal
            product_name={this.props.product_name}
            question={this.props.question.question_body}
            missing={this.state.missingFields}
            submit={this.handleSubmitAnswer}
            close={this.closeModal}
          />
        }
        <div className="question-title">
          <p className="question-body">Q: {this.props.question.question_body}</p>
          <div className="question-links">
            <p >Helpful?&nbsp;
              <a className="question-link" onClick={this.handleHelpfulnessVote}>Yes</a>
              &nbsp;({this.state.helpfulnessVoteCount})&nbsp;|&nbsp;
              <a className="question-link" onClick={this.handleAddAnswer}>Add Answer</a>
            </p>
          </div>
        </div>
        <AnswerList
          answers={this.props.question.answers}
        />
      </div>
    );
  }
}

Question.propTypes = {
  question: PropTypes.object,
  product_name: PropTypes.string
};

export default Question;