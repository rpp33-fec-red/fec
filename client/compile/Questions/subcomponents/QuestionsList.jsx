/*global React */
/*eslint no-undef: "error"*/
import PropTypes from 'prop-types';
import Question from './Question.jsx';


class QuestionsList extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    let questions = this.props.questions;

    if (questions.length) {
      questions = questions.slice(0, this.props.displayed);

      return(
        <div className='questions-list'>
          {questions.map((question) =>
            <Question
              key={question.question_id}
              question={question}
              product_name={this.props.product_name}
            />
          )}
        </div>
      );
    } else {
      return null;
    }
  }
}

QuestionsList.propTypes = {
  questions: PropTypes.any,
  displayed: PropTypes.number,
  product_name: PropTypes.string
};

export default QuestionsList;