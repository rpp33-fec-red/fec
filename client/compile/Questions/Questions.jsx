import React from 'react';
import axios from 'axios';
import PropTypes from 'prop-types';
import SearchBar from './subcomponents/SearchBar.jsx';
import QuestionsList from './subcomponents/QuestionsList.jsx';
import QuestionButtons from './subcomponents/QuestionButtons.jsx';
import {sampleData} from './sampleData.js';
import './questionsStyles.scss';

class QuestionsWidget extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      query: '',
      queriedQuestions: sampleData.results,
      maxQuestionsDisplayed: 2,
      allQuestionsDisplayed: false,
      questionsData: sampleData.results
    };
    this.handleSearch = this.handleSearch.bind(this);
    this.handleDisplayMoreQuestions = this.handleDisplayMoreQuestions.bind(this);
    this.queryQuestions = this.queryQuestions.bind(this);
  }

  componentDidMount() {
    let request = {
      endpoint: '/qa/questions',
      params: {
        product_id: this.props.product_id,
        page: 1,
        count: 20
      }
    };
    axios.post('/getQuestions', request)
      .then((response) => {
        if (response.data.results) {
          const sorted = response.data.results.sort((a, b) => {
            return b.question_helpfulness - a.question_helpfulness;
          });
          if (sorted.length < 3) {
            this.setState({
              allQuestionsDisplayed: true,
              queriedQuestions: sorted,
              questionsData: sorted
            });
          } else {
            this.setState({
              allQuestionsDisplayed: false,
              queriedQuestions: sorted,
              questionsData: sorted
            });
          }
        } else {
          this.setState({
            allQuestionsDisplayed: true,
            questionsData: [],
            queriedQuestions: []
          });
        }
      })
      .catch((err) => {
        throw err;
      });
  }

  handleDisplayMoreQuestions() {
    this.setState({
      maxQuestionsDisplayed: this.state.maxQuestionsDisplayed + 2
    }, () => {
      if (this.state.queriedQuestions.length <= this.state.maxQuestionsDisplayed) {
        this.setState({
          allQuestionsDisplayed: true
        });
      }
    });
  }

  handleSearch(event) {
    event.preventDefault();
    let query = event.target.value;
    if (query.length > 2) {
      this.setState({
        query: query
      }, function() {
        let questions = this.queryQuestions(this.state.questionsData);
        this.setState({
          queriedQuestions: questions
        });
      });
    } else if (this.state.query !== '') {
      this.setState({
        query: '',
        queriedQuestions: this.state.questionsData
      });
    }
  }

  queryQuestions(questions) {
    let questionsSearched = [];
    for (const question of questions) {
      if (question.question_body.indexOf(this.state.query) !== -1) {
        questionsSearched.push(question);
      }
    }
    return questionsSearched;
  }

  render() {
    return (
      <div className="questions-widget">
        <p className="questions-title">QUESTIONS &amp; ANSWERS</p>
        <SearchBar search={this.handleSearch}/>
        <QuestionsList
          questions={this.state.queriedQuestions}
          displayed={this.state.maxQuestionsDisplayed}
          scroll={this.state.allQuestionsDisplayed}
          product_name={this.props.product_name}
        />
        <QuestionButtons
          questions={this.state.questionsData}
          allQuestionsDisplayed={this.state.allQuestionsDisplayed}
          displayMore={this.handleDisplayMoreQuestions}
          product_id={this.props.product_id}
          product_name={this.props.product_name}
        />
      </div>
    );
  }
}

QuestionsWidget.propTypes = {
  getQuestions: PropTypes.func,
  product_id: PropTypes.string,
  product_name: PropTypes.string
};

export default QuestionsWidget;