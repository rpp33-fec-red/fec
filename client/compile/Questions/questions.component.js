import React from 'react';
import './questions.scss';
import SearchBar from './subcomponents/SearchBar.jsx';
import QuestionsList from './subcomponents/QuestionsList.jsx';
import MoreQuestions from './subcomponents/MoreQuestions.jsx';
import AddQuestion from './subcomponents/AddQuestion.jsx';


class Questions extends React.component {
  constructor(props) {
    super(props);
    this.state = {
      query: '',
      expanded: 0
    };
  }

  render() {
    return (
      <div className="questionsWidget">
        <h6>QUESTIONS &amp; ANSWERS</h6>
        <SearchBar />
        <div className="questionsList">
          <QuestionsList expanded={this.state.expanded}/>
        </div>
        <div className="questionButtons">
          <MoreQuestions />
          <div></div>
          <AddQuestion />
        </div>
      </div>
    )
  }
}

export default Questions;