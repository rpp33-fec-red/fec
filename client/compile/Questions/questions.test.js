import React from 'react';
import TestRenderer from 'react-test-renderer';
import Questions from './Questions.jsx';
import QuestionButtons from './subcomponents/QuestionButtons.jsx';
import SearchBar from './subcomponents/SearchBar.jsx';
import Question from './subcomponents/Question.jsx';
import QuestionsList from './subcomponents/QuestionsList.jsx';
import Answer from './subcomponents/Answer.jsx';
import AnswerList from './subcomponents/AnswerList.jsx';
import { sampleData, answer } from './sampleData.js';

//https://jestjs.io/docs/getting-started

describe('Questions widget component', () => {
  test('loads and displays questions widget component', () => {
    var questionsTree = TestRenderer.create(<Questions />).toJSON();
    expect(questionsTree).toMatchSnapshot();
  });
});

describe('QuestionButtons component', () => {
  test('loads and displays Question Buttons comoponent', () => {
    var questionButtonsTree = TestRenderer.create(<QuestionButtons allQuestionsDisplayed={false} />).toJSON();
    expect(questionButtonsTree).toMatchSnapshot();
  });
});

describe('SearchBar component', () => {
  test('loads and displays SearchBar component', () => {
    var searchBarTree = TestRenderer.create(<SearchBar />).toJSON();
    expect(searchBarTree).toMatchSnapshot();
  });
});

describe('QuestionsList component', () => {
  test('loads and displays QuestionsList component', () => {
    var questionsListTree = TestRenderer.create(<QuestionsList questions={sampleData}/>).toJSON();
    expect(questionsListTree).toMatchSnapshot();
  });
});

describe('Question component', () => {
  test('loads and displays Question component', () => {
    var questionTree = TestRenderer.create(<Question question={sampleData.results[0]}/>).toJSON();
<<<<<<< HEAD
=======
<<<<<<< HEAD
=======

>>>>>>> d9c643e05bdea633ff8be31ce2b6cf9abfbf5bc8
>>>>>>> 1272cdcb3db7b29ece26c22d74cb0c249cf1c294
    expect(questionTree).toMatchSnapshot();
  });
});

describe('AnswerList component', () => {
  test('loads and displays AnswerList component', () => {
    var answerListTree = TestRenderer.create(<AnswerList answers={sampleData.results[0].answers}/>).toJSON();

    expect(answerListTree).toMatchSnapshot();
  });
});

describe('Answer component', () => {
  test('loads and displays Answer component', () => {
    var answerTree = TestRenderer.create(<Answer answer={answer}/>).toJSON();
    expect(answerTree).toMatchSnapshot();
  });
});
