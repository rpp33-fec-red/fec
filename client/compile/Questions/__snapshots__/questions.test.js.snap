// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAnswerModal component loads and displays Answer component 1`] = `
<div
  className="questions-modals"
  id="add-answer-modal"
>
  <div
    className="questions-modals-content"
  >
    <a
      className="close-modal"
    >
      X
    </a>
    <h2>
      Submit your Answer
    </h2>
    <h3>
      : 
    </h3>
    <form
      className="add-answer-form"
    >
      <div
        className="modal-form"
      >
        <label>
          Your Answer*
        </label>
        <textarea
          id="new-answer-field"
          maxLength="1000"
          name="answer"
          type="text"
        />
      </div>
      <div
        className="modal-form"
      >
        <label>
          What is your nickname*
        </label>
        <input
          maxLength="60"
          name="nickname"
          placeholder="Example: jack543!"
          type="text"
        />
        <p>
          For privacy reasons, do not use your full name or email address
        </p>
      </div>
      <div
        className="modal-form"
      >
        <label>
          Your email*
        </label>
        <input
          maxLength="60"
          name="email"
          placeholder="Example: jack@email.com"
          type="email"
        />
        <p>
          For authentication reasons, you will not be emailed
        </p>
      </div>
      <div
        className="modal-form"
      >
        <input
          name="photos"
          onClick={[Function]}
          type="button"
          value="Upload Photos"
        />
      </div>
      <div
        className="modal-form"
      >
        <input
          type="submit"
          value="Submit answer"
        />
      </div>
    </form>
  </div>
</div>
`;

exports[`AddQuestionModal component loads and displays Answer component 1`] = `
<div
  className="questions-modals"
  id="add-question-modal"
>
  <div
    className="questions-modals-content"
  >
    <a
      className="close-modal"
    >
      X
    </a>
    <h2>
      Ask Your Question
    </h2>
    <h3>
      About the 
    </h3>
    <form
      className="add-question-form"
    >
      <div
        className="modal-form"
      >
        <label>
          Your Question*
        </label>
        <textarea
          id="new-question-field"
          maxLength="1000"
          name="question"
          type="text"
        />
      </div>
      <div
        className="modal-form"
      >
        <label>
          What is your nickname*
        </label>
        <input
          maxLength="60"
          name="nickname"
          placeholder="Example: jackson11!"
          type="text"
        />
        <p>
          For privacy reasons, do not use your full name or email address
        </p>
      </div>
      <div
        className="modal-form"
      >
        <label>
          Your email*
        </label>
        <input
          maxLength="60"
          name="email"
          placeholder="Why did you like the product or not?"
          type="email"
        />
        <p>
          For authentication reasons, you will not be emailed
        </p>
      </div>
      <div
        className="modal-form"
      >
        <input
          type="submit"
          value="Submit question"
        />
      </div>
    </form>
  </div>
</div>
`;

exports[`Answer component loads and displays Answer component 1`] = `
<div
  className="answer"
>
  <p
    className="answer-title"
  >
    We are selling it here without any markup from the middleman!
  </p>
  <div
    className="answer-links"
  >
    <p>
      by
      <span
        className="answerer-seller"
      >
         
        Seller
      </span>
      , 
      August, 17, 2018
        |  Helpful? 
      <a
        className="answer-link"
        onClick={[Function]}
      >
        Yes
      </a>
       (
      4
      )  |  
      <a
        className="answer-link"
        onClick={[Function]}
      >
        Report
      </a>
    </p>
  </div>
</div>
`;

exports[`AnswerList component loads and displays AnswerList component 1`] = `
<div
  className="answers"
>
  <p
    className="a"
  >
    A: 
  </p>
  <div
    className="answer-list"
  >
    <div
      className="answer"
    >
      <p
        className="answer-title"
      >
        We are selling it here without any markup from the middleman!
      </p>
      <div
        className="answer-links"
      >
        <p>
          by
          <span
            className="answerer-seller"
          >
             
            Seller
          </span>
          , 
          August, 17, 2018
            |  Helpful? 
          <a
            className="answer-link"
            onClick={[Function]}
          >
            Yes
          </a>
           (
          4
          )  |  
          <a
            className="answer-link"
            onClick={[Function]}
          >
            Report
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Question component loads and displays Question component 1`] = `
<div
  className="question"
>
  <div
    className="questionTitle"
  >
    <p
      className="question-body"
    >
      Q: 
      Why is this product cheaper here than other sites?
    </p>
    <div
      className="questionLinks"
    >
      <p>
        Helpful? 
        <a
          className="question-link"
          onClick={[Function]}
        >
          Yes
        </a>
         (
        4
        ) | 
        <a
          className="question-link"
          onClick={[Function]}
        >
          Add Answer
        </a>
      </p>
    </div>
  </div>
  <div
    className="answers"
  >
    <p
      className="a"
    >
      A: 
    </p>
    <div
      className="answer-list"
    >
      <div
        className="answer"
      >
        <p
          className="answer-title"
        >
          We are selling it here without any markup from the middleman!
        </p>
        <div
          className="answer-links"
        >
          <p>
            by
            <span
              className="answerer-seller"
            >
               
              Seller
            </span>
            , 
            August, 17, 2018
              |  Helpful? 
            <a
              className="answer-link"
              onClick={[Function]}
            >
              Yes
            </a>
             (
            4
            )  |  
            <a
              className="answer-link"
              onClick={[Function]}
            >
              Report
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`QuestionButtons component loads and displays Question Buttons comoponent 1`] = `
<div
  className="question-buttons"
>
  <button>
    MORE ANSWERED QUESTIONS
  </button>
  <button
    onClick={[Function]}
  >
    ADD A QUESTION +
  </button>
</div>
`;

exports[`Questions widget component loads and displays questions widget component 1`] = `
<div
  className="questions-widget"
>
  <h6>
    QUESTIONS & ANSWERS
  </h6>
  <div
    className="search-bar"
  >
    <input
      id="questions-search-bar"
      onChange={[Function]}
      placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..."
      type="search"
    />
  </div>
  <div
    className="questions-list"
  >
    <div
      className="question"
    >
      <div
        className="questionTitle"
      >
        <p
          className="question-body"
        >
          Q: 
          Why is this product cheaper here than other sites?
        </p>
        <div
          className="questionLinks"
        >
          <p>
            Helpful? 
            <a
              className="question-link"
              onClick={[Function]}
            >
              Yes
            </a>
             (
            4
            ) | 
            <a
              className="question-link"
              onClick={[Function]}
            >
              Add Answer
            </a>
          </p>
        </div>
      </div>
      <div
        className="answers"
      >
        <p
          className="a"
        >
          A: 
        </p>
        <div
          className="answer-list"
        >
          <div
            className="answer"
          >
            <p
              className="answer-title"
            >
              We are selling it here without any markup from the middleman!
            </p>
            <div
              className="answer-links"
            >
              <p>
                by
                <span
                  className="answerer-seller"
                >
                   
                  Seller
                </span>
                , 
                August, 17, 2018
                  |  Helpful? 
                <a
                  className="answer-link"
                  onClick={[Function]}
                >
                  Yes
                </a>
                 (
                4
                )  |  
                <a
                  className="answer-link"
                  onClick={[Function]}
                >
                  Report
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="question"
    >
      <div
        className="questionTitle"
      >
        <p
          className="question-body"
        >
          Q: 
          How long does it last?
        </p>
        <div
          className="questionLinks"
        >
          <p>
            Helpful? 
            <a
              className="question-link"
              onClick={[Function]}
            >
              Yes
            </a>
             (
            2
            ) | 
            <a
              className="question-link"
              onClick={[Function]}
            >
              Add Answer
            </a>
          </p>
        </div>
      </div>
      <div
        className="answers"
      >
        <p
          className="a"
        >
          A: 
        </p>
        <div
          className="answer-list"
        >
          <div
            className="answer"
          >
            <p
              className="answer-title"
            >
              9 lives
            </p>
            <div
              className="answer-links"
            >
              <p>
                by
                <span
                  className="answerer-other"
                >
                   
                  iluvdogz
                </span>
                , 
                November, 11, 2019
                  |  Helpful? 
                <a
                  className="answer-link"
                  onClick={[Function]}
                >
                  Yes
                </a>
                 (
                31
                )  |  
                <a
                  className="answer-link"
                  onClick={[Function]}
                >
                  Report
                </a>
              </p>
            </div>
          </div>
          <div
            className="answer"
          >
            <p
              className="answer-title"
            >
              Some of the seams started splitting the first time I wore it!
            </p>
            <div
              className="answer-links"
            >
              <p>
                by
                <span
                  className="answerer-other"
                >
                   
                  sillyguy
                </span>
                , 
                November, 27, 2019
                  |  Helpful? 
                <a
                  className="answer-link"
                  onClick={[Function]}
                >
                  Yes
                </a>
                 (
                6
                )  |  
                <a
                  className="answer-link"
                  onClick={[Function]}
                >
                  Report
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="question-buttons"
  >
    <button
      onClick={[Function]}
    >
      MORE ANSWERED QUESTIONS
    </button>
    <button
      onClick={[Function]}
    >
      ADD A QUESTION +
    </button>
  </div>
</div>
`;

exports[`QuestionsList component loads and displays QuestionsList component 1`] = `null`;

exports[`SearchBar component loads and displays SearchBar component 1`] = `
<div
  className="search-bar"
>
  <input
    id="questions-search-bar"
    placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..."
    type="search"
  />
</div>
`;

exports[`UploadPhotosModal component loads and displays Answer component 1`] = `
<div
  className="questions-modals"
  id="upload-photo-modal"
>
  <div
    className="questions-modals-content"
  >
    <a
      className="close-window"
    >
      X
    </a>
    <h3>
      Upload Photos for 
    </h3>
    <form>
      <div>
        <label
          htmlFor="photo-upload"
        >
          Choose photo to upload
        </label>
        <input
          accept="image/png, image/jpeg"
          id="upload-answer-photo"
          name="photoUpload"
          type="file"
        />
      </div>
      <div>
        <input
          type="submit"
          value="Upload Photo"
        />
      </div>
    </form>
    <div
      className="selected-photos"
    />
  </div>
</div>
`;
